pkg/deploy/service_accounts.go

```diff
@@ -49,43 +61,27 @@ func (s *ServiceAccounts) fetchServiceAccount(nsName string, saName string) (str
 		return "", fmt.Errorf("Internal inconsistency: Expected service account name to not be empty")
 	}
 
-	sa, err := s.coreClient.CoreV1().ServiceAccounts(nsName).Get(context.Background(), saName, metav1.GetOptions{})
+	expiration := int64(tokenExpiration.Seconds())
+	t, err := s.tokenManager.GetServiceAccountToken(nsName, saName, &authenticationv1.TokenRequest{
+		Spec: authenticationv1.TokenRequestSpec{
+			ExpirationSeconds: &expiration,
+		},
+	})
 	if err != nil {
-		return "", fmt.Errorf("Getting service account: %s", err)
+		return "", fmt.Errorf("Get service account token: %s", err)
 	}
 
-	for _, secretRef := range sa.Secrets {
-		secret, err := s.coreClient.CoreV1().Secrets(nsName).Get(context.Background(), secretRef.Name, metav1.GetOptions{})
+	if len(s.caCert) == 0 {
```

Comment:
needs to be behind mutex since parallel calls to Find may/will happen. alternatively can move it into constructor and record err (to be returned from Find) to avoid rereading cert.